File: backend/internal/domain/lesson.go
package domain

import "time"

// Status types and constants now defined in status.go (shared with all modules)

type Course struct {
	ID                string       `json:"id"`
	Code              string       `json:"code"`
	Name              string       `json:"name"`
	Type              string       `json:"type"`
	Category          string       `json:"category"`
	MajorID           *string      `json:"majorId,omitempty"`
	MajorName         *string      `json:"majorName,omitempty"`
	TeacherID         *string      `json:"teacherId,omitempty"`
	IndustryID        *string      `json:"industryId,omitempty"`
	IndustryName      *string      `json:"industryName,omitempty"`
	Version           *string      `json:"version,omitempty"`
	OnlineHours       *float64     `json:"onlineHours,omitempty"`
	OfflineHours      *float64     `json:"offlineHours,omitempty"`
	OnlineWeight      *float64     `json:"onlineWeight,omitempty"`
	OfflineWeight     *float64     `json:"offlineWeight,omitempty"`
	Semester          *string      `json:"semester,omitempty"`
	ClassName         *string      `json:"className,omitempty"`
	Status            CourseStatus `json:"status"`
	CoverColor        *string      `json:"coverColor,omitempty"`
	CoverImage        *string      `json:"coverImage,omitempty"`
	CourseTag         *string      `json:"courseTag,omitempty"`
	Difficulty        *int         `json:"difficulty,omitempty"`
	Description       *string      `json:"description,omitempty"`
	KnowledgePointIds JSONSlice    `json:"knowledgePointIds,omitempty"`
	AbilityPointIds   JSONSlice    `json:"abilityPointIds,omitempty"`
	ResourceIds       JSONSlice    `json:"resourceIds,omitempty"`
	CreatorID         string       `json:"creatorId"`
	CoCreatorIds      JSONSlice    `json:"coCreatorIds,omitempty"`
	BatchID           *string      `json:"batchId,omitempty"`
	BatchName         *string      `json:"batchName,omitempty"`
	EvalData          JSONMap      `json:"evalData,omitempty"`
	NodeCount         int          `json:"nodeCount"`
	ResourceCount     int          `json:"resourceCount"`
	ViewCount         int          `json:"viewCount"`
	StudyCount        int          `json:"studyCount"`
	CreatedAt         time.Time    `json:"createdAt"`
	UpdatedAt         time.Time    `json:"updatedAt"`
}

type KnowledgePoint struct {
	ID                string    `json:"id"`
	Name              string    `json:"name"`
	Code              *string   `json:"code,omitempty"`
	Description       *string   `json:"description,omitempty"`
	Category          *string   `json:"category,omitempty"`
	Linked            bool      `json:"linked"`
	GranularLessonIds JSONSlice `json:"granularLessonIds,omitempty"`
	CreatorID         *string   `json:"creatorId,omitempty"`
	SourceType        *string   `json:"sourceType,omitempty"`
	SourceID          *string   `json:"sourceId,omitempty"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
}

type NodeQuiz struct {
	ID        string `json:"id"`
	NodeID    string `json:"nodeId"`
	Title     string `json:"title"`
	Type      string `json:"type"`
	TimeLimit *int   `json:"timeLimit,omitempty"`
}

type NodeQuizQuestion struct {
	ID        string  `json:"id"`
	QuizID    string  `json:"quizId"`
	Type      string  `json:"type"`
	Question  string  `json:"question"`
	Options   JSONMap `json:"options,omitempty"`
	Answer    *string `json:"answer,omitempty"`
	Score     float64 `json:"score"`
	SortOrder int     `json:"sortOrder"`
}

type NodeHomework struct {
	ID             string     `json:"id"`
	NodeID         string     `json:"nodeId"`
	Title          string     `json:"title"`
	Requirement    *string    `json:"requirement,omitempty"`
	NeedAttachment bool       `json:"needAttachment"`
	Deadline       *time.Time `json:"deadline,omitempty"`
}

type HybridNodeModule struct {
	ID        string  `json:"id"`
	NodeID    string  `json:"nodeId"`
	ModuleKey string  `json:"moduleKey"`
	Mode      string  `json:"mode"`
	Data      JSONMap `json:"data"`
}

type LessonBatch struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	Code        *string           `json:"code,omitempty"`
	OrgNodeID   *string           `json:"orgNodeId,omitempty"`
	MajorID     *string           `json:"majorId,omitempty"`
	MajorName   *string           `json:"majorName,omitempty"`
	WorkflowID  *string           `json:"workflowId,omitempty"`
	Status      LessonBatchStatus `json:"status"`
	CourseCount *int              `json:"courseCount,omitempty"`
	CreatedAt   time.Time         `json:"createdAt"`
	UpdatedAt   time.Time         `json:"updatedAt"`
}

type LessonBehaviorRecord struct {
	ID               string    `json:"id"`
	CourseID         string    `json:"courseId"`
	StudentUserID    string    `json:"studentUserId"`
	StudentName      string    `json:"studentName"`
	RecordDate       string    `json:"recordDate"`
	Attendance       string    `json:"attendance"`
	QuizScore        *float64  `json:"quizScore,omitempty"`
	InteractionCount int       `json:"interactionCount"`
	PraiseCount      int       `json:"praiseCount"`
	RushCorrectCount int       `json:"rushCorrectCount"`
	RushAvgTimeSec   *int      `json:"rushAvgTimeSec,omitempty"`
	CreatedAt        time.Time `json:"createdAt"`
	UpdatedAt        time.Time `json:"updatedAt"`
}

type NodeResource struct {
	ID          string    `json:"id"`
	NodeID      string    `json:"nodeId"`
	Name        string    `json:"name"`
	Type        string    `json:"type"`
	URL         *string   `json:"url,omitempty"`
	Description *string   `json:"description,omitempty"`
	Size        *int      `json:"size,omitempty"`
	UploadedBy  *string   `json:"uploadedBy,omitempty"`
	UploadedAt  time.Time `json:"uploadedAt"`
}

File: packages/shared-types/src/lesson.ts
export interface Course {
  id: string
  code: string
  name: string
  type: "system" | "granular" | "hybrid"
  category: string
  majorId?: string
  majorName?: string
  description?: string
  teacherId?: string
  industryId?: string
  industryName?: string
  version?: string
  onlineHours?: number
  offlineHours?: number
  onlineWeight?: number
  offlineWeight?: number
  semester?: string
  className?: string
  status: "draft" | "pending" | "approved" | "rejected" | "published" | "archived"
  coverColor?: string
  coverImage?: string
  courseTag?: string
  creatorId: string
  coCreatorIds: string[]
  batchId?: string
  batchName?: string
  nodeCount: number
  resourceCount: number
  studyCount: number
  createdAt: string
  updatedAt: string
}

export interface KnowledgePoint {
  id: string
  name: string
  code?: string
  description?: string
  linked: boolean
  granularLessonIds: string[]
  creatorId?: string
  createdAt: string
  updatedAt: string
}

export interface SystemCourseNode {
  id: string
  courseId: string
  parentId?: string
  name: string
  sortOrder: number
  refType: "normal" | "original" | "resource"
  sourceId?: string
  sourceName?: string
  teachingGoals?: string
  duration?: number
  knowledgePointIds: string[]
  resourceIds: string[]
  status: string
  createdAt: string
  updatedAt: string
}

export interface NodeQuiz {
  id: string
  nodeId: string
  title: string
  type: "paper" | "question_bank"
  timeLimit?: number
}

export interface NodeQuizQuestion {
  id: string
  quizId: string
  type: "single" | "multiple" | "judge" | "essay"
  question: string
  options?: Record<string, any>
  answer?: string
  score: number
  sortOrder: number
}

export interface NodeHomework {
  id: string
  nodeId: string
  title: string
  requirement?: string
  needAttachment: boolean
  deadline?: string
}

export interface HybridNodeModule {
  id: string
  nodeId: string
  moduleKey: string
  mode: "online" | "offline"
  data: Record<string, any>
}

export interface NodeResource {
  id: string
  nodeId: string
  name: string
  type: string
  url: string
  size?: number
}

export interface CourseKnowledgeBinding {
  id: string
  courseId: string
  knowledgePointId: string
  bindType: "course" | "node"
  sourceId?: string
}

export interface LessonBatch {
  id: string
  tenantId?: string
  name: string
  code?: string
  orgNodeId?: string
  majorId?: string
  majorName?: string // Deprecated
  workflowId?: string
  status: "open" | "closed"
  courseCount?: number
  createdAt: string
  updatedAt: string
}

// ==================== Admin component local types (migrated from lib/mock-data-lesson) ====================

export type ResourceType = "document" | "video" | "link" | "file"

export interface Resource {
  id: string
  name: string
  type: ResourceType
  url: string
  size?: string
}

export type LessonQuestionType = "single" | "multiple" | "judgment"

export interface QuestionItem {
  id: string
  type: LessonQuestionType
  content: string
  options?: string[]
  answer: string | string[]
  score: number
}

export interface ObjectiveConfig {
  questions: QuestionItem[]
  totalScore: number
}

export interface RubricLevel {
  id: string
  name: string
  minScore: number
  maxScore: number
  description: string
  color: string
}

export interface RubricPoint {
  id: string
  name: string
  weight: number
  maxScore: number
  levels: RubricLevel[]
}

export interface SubjectiveConfig {
  rubricPoints: RubricPoint[]
  synthesisRule: "sum" | "weighted"
}

export interface GradeMapping {
  id: string
  grade: string
  minScore: number
  maxScore: number
  color: string
  remark?: string
}

export interface KnowledgePointItem {
  id: string
  name: string
  code?: string
  description?: string
  linked: boolean
  granularLessons?: string[]
}

export interface EvalPoint {
  id: string
  name: string
  desc: string
  subType?: string
  types?: string[]
  knowledgePointIds?: string[]
  abilityPointIds?: string[]
  scoringMethod?: "score" | "level" | "rubric"
  gradeMapping?: GradeMapping[]
  weight?: number
}

export interface LessonBehaviorRecord {
  id: string
  courseId: string
  studentUserId: string
  recordDate: string
  attendance: "present" | "late" | "absent"
  quizScore?: number
  interactionCount: number
  praiseCount: number
  rushCorrectCount: number
  rushAvgTimeSec?: number
  createdAt: string
  updatedAt: string
}

export interface LessonBehaviorAggregate {
  signIn: {
    total: number
    present: number
    late: number
    absent: number
    rate: number
  }
  signInDaily: { date: string; present: number; late: number; absent: number }[]
  quizResults: { id: string; name: string; avgScore: number; passRate: number; count: number }[]
  rushAnswerRanking: { rank: number; name: string; correctCount: number; avgTime: string; badge: string }[]
  classInteraction: { name: string; active: number; total: number }[]
  attendanceRateData: { name: string; rate: number }[]
  studentDetails: { name: string; attendance: number; quizAvg: number; interaction: number; praise: number }[]
}
